presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/s3select/S3SelectTestHelper.java

```diff
@@ -0,0 +1,239 @@
+import com.facebook.presto.hive.NodeVersion;
+import com.facebook.presto.hive.TableParameterCodec;
+import com.facebook.presto.hive.authentication.NoHdfsAuthentication;
+import com.facebook.presto.hive.datasink.OutputStreamDataSinkFactory;
+import com.facebook.presto.hive.metastore.HivePartitionMutator;
+import com.facebook.presto.hive.metastore.thrift.BridgingHiveMetastore;
+import com.facebook.presto.hive.metastore.thrift.HiveCluster;
+import com.facebook.presto.hive.metastore.thrift.TestingHiveCluster;
+import com.facebook.presto.hive.metastore.thrift.ThriftHiveMetastore;
+import com.facebook.presto.hive.s3.HiveS3Config;
+import com.facebook.presto.hive.s3.PrestoS3ConfigurationUpdater;
+import com.facebook.presto.hive.s3.S3ConfigurationUpdater;
+import com.facebook.presto.spi.connector.ConnectorPageSourceProvider;
+import com.facebook.presto.spi.connector.ConnectorSplitManager;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.net.HostAndPort;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+import java.util.concurrent.ExecutorService;
+
+import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
+import static com.facebook.presto.hive.HiveTestUtils.FILTER_STATS_CALCULATOR_SERVICE;
+import static com.facebook.presto.hive.HiveTestUtils.FUNCTION_AND_TYPE_MANAGER;
+import static com.facebook.presto.hive.HiveTestUtils.FUNCTION_RESOLUTION;
+import static com.facebook.presto.hive.HiveTestUtils.ROW_EXPRESSION_SERVICE;
+import static com.facebook.presto.hive.HiveTestUtils.getDefaultHiveBatchPageSourceFactories;
+import static com.facebook.presto.hive.HiveTestUtils.getDefaultHiveSelectivePageSourceFactories;
+import static com.facebook.presto.hive.HiveTestUtils.getDefaultS3HiveRecordCursorProvider;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.util.concurrent.MoreExecutors.listeningDecorator;
+import static com.google.common.util.concurrent.MoreExecutors.newDirectExecutorService;
+import static java.lang.String.format;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static org.testng.util.Strings.isNullOrEmpty;
+
+public class S3SelectTestHelper
+{
+    private HdfsEnvironment hdfsEnvironment;
+    private LocationService locationService;
+    private TestingHiveMetastore metastoreClient;
+    private HiveMetadataFactory metadataFactory;
+    private HiveTransactionManager transactionManager;
+    private ConnectorSplitManager splitManager;
+    private ConnectorPageSourceProvider pageSourceProvider;
+
+    private ExecutorService executor;
+    private HiveClientConfig config;
+    private CacheConfig cacheConfig;
+    private MetastoreClientConfig metastoreClientConfig;
+
+    @BeforeClass
+    public void setUp()
+    {
+        executor = newCachedThreadPool(daemonThreadsNamed("hive-%s"));
+    }
+
+    @AfterClass(alwaysRun = true)
```

Comment:
These annotations only work when run on the current class as a test. This appears to just be a utility class used from other tests, so those will need to have setup / teardown methods of their own with these annotations in order to be called.

Reply:
Correct, thanks for catching this! I mixed things up during rebasing & moving from the test class to the helper class.